keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepository.kt

```diff
@@ -108,6 +110,22 @@ interface ResourceRepository : PeriodicallyCheckedRepository<Resource<out Resour
    */
   fun deleteByApplication(application: String): Int
 
+  /**
+   * Retrieves the history of persisted events for [application].
+   *
+   * @param application the name of the application.
+   * @param limit the maximum number of events to return.
+   */
+  fun applicationEventHistory(application: String, limit: Int = DEFAULT_MAX_EVENTS): List<ApplicationEvent>
```

Comment:
We don't have an `ApplicationRepository` since we don't store application metadata, so I decided to add these here for now. Let me know if there's a better place.